### PR TITLE
Made compatible with “Dark mode” setting from the browser

### PR DESCRIPTION
**Pull request by doedje**
Use of variables on :root in samltrace.css for all the colors and button-icons, which will be overwritten in dark-mode, if needed.

I tried to stay as much to the original as possible for the 'light mode'. But I did bring down the number of different shades of grey that were in use. Notable changes are: a) the model is little bit more subtle and b) the error-message in the import dialog I changed to something that works equally fine on both white and black backgrounds.

### CODE REVIEW

**Comment by khlr** [comment]
Hello Remy!

Thank you so much for this! A really great and helpful contribution. 😊
I think all dark mode users will be very grateful to you.

From what I've seen everything seems to work and look really fine. So for my part, there is nothing that stands in the way of a merge.

There is only one small thing you could possibly think about: Currently the background is 100% black. My personal feeling is that this creates a very hard contrast. Many dark modes use a just very dark gray instead, which I find more pleasing to the eye. Firefox uses #1C1B22 for example.


*1: As is from your PR:*

image

*2: With #1C1B22:*

image

*3: Softened colors due to import overlay:*

image

Again, this is just my subjective opinion; I would choose an adjusted version of variant 3. To me the combination of a softened background _and_ softened foreground text colors seems very pleasant.

What's your opinion?
Do you feel like playing with the colors a bit more? If not, it is also completely ok. As I said, I think all dark mode users will be very grateful about anything that is not bright :-)

**Comment by tvdijen** [comment]
I would also be in favor of a little-less-than-black! That import-dialog could use some more TLC though.. The [Import] button is really hard to read because the rest of the dialog is so black..

**Review by khlr — APPROVED** [review]
Hello Remy!

Thank you so much for this! A really great and helpful contribution. 😊
I think all dark mode users will be very grateful to you.

From what I've seen everything seems to work and look really fine. So for my part, there is nothing that stands in the way of a merge.

There is only one small thing you could possibly think about: Currently the background is 100% black. My personal feeling is that this creates a very hard contrast. Many dark modes use a just very dark gray instead, which I find more pleasing to the eye. Firefox uses #1C1B22 for example.


*1: As is from your PR:*

image

*2: With #1C1B22:*

image

*3: Softened colors due to import overlay:*

image

Again, this is just my subjective opinion; I would choose an adjusted version of variant 3. To me the combination of a softened background _and_ softened foreground text colors seems very pleasant.

What's your opinion?
Do you feel like playing with the colors a bit more? If not, it is also completely ok. As I said, I think all dark mode users will be very grateful about anything that is not bright :-)

**Comment by doedje** [comment]
Thanx for the feedback, I totally agree on the little-less-than-black! I'll colorpick the colors from Firefox's Web developer tools, for both dark and light mode.
(the import button in the screenshot is indeed hard to read, but it is in its disabled state in the screenshot, but I'll look into that.
Are you okay with what I did to the modal?

**Comment by doedje** [comment]
I pushed a new commit in which I selected all colors from the Firefox Web Developer Tool, both dark and light mode. Same goes for the error message in the import pane. I also took the liberty of changing the font-family for #request-info-content, in my case it now uses the same font as in the Firefox Web Developer Tool, resulting in much better readable info.

**Comment by doedje** [comment]
Please merge yet, I am also color picking the colors for the syntax highlighting from the devtools. Will finish that tomorrow and do another push

**Comment by doedje** [comment]
This should be it, for now! =]

**Comment by khlr** [comment]
Nice, I really like it 😊

Just one thing, that you _could_ adjust, if you like to:
Requests without responses have the default color (`#d7d7db`) which appears very bright (almost white against the dark background). Given that these requests are the least important most of the times, I'd prefer to darken them. Something like the default text color for the bright theme (`#737373`) seems quite nice to me. What do you think?

**Comment by doedje** [comment]
I changed it as per your suggestion. But shouldn't the "Filter resources" button take care of all those uninteresting request in the first place? I have the feeling this filtering was 'better' in the past and it is now (last couple of years) showing me much more uninteresting request than it used to.

**Comment by khlr** [comment]
> [...] But shouldn't the "Filter resources" button take care of all those uninteresting request in the first place? 

Those responseless requests and requests for resources are two different things. The latter are usually always not of interest - hence the filter-away-feature.
Responeless requests, however, can actually be very interesting! Take a look at my last screenshot - the POST to sp.example.com. There was no response, because there's nothing at all. But in a real situation this could be a valid thing. Consider the service at sp.example.com by mistake just isn't running. You probably would the request to appear in the list anyway.

> I have the feeling this filtering was 'better' in the past and it is now (last couple of years) showing me much more uninteresting request than it used to.

There are quite a lot of reqeusts to resources that don't receive a response. So it's a combination of both. Why does this happen? I don't know. Maybe related to caching?
If the filter-resources-feature was a bit smarter it probably detected those responseless resource requests anyway. Maybe something to enhance...

File: src/resources/css/samltrace.css [hunk 1]
```diff
@@ -178,7 +178,7 @@ a.tab.selected:hover {
   display: flex;
   white-space: nowrap;
   border-bottom: 1px solid var(--color-border);
-  color: var(--color-text);
+  color: var(--color-syntax-comment);
```
[the remark this answers is not in the document]
Comment (khlr) [reply src/resources/css/samltrace.css:181]: I don't know if you're willing to change this, too. If not - absolutely ok 😊 I think we all appreciate the work you did. Thank you!

[the remark this answers is not in the document]
Comment (doedje) [reply src/resources/css/samltrace.css]: yeah, I was being lazy... changed it to `--color-line-text`

**Comment by khlr** [comment]
Merged 😊 Thank you!

**Pull request merged**